stable/nginx-ingress/templates/default-backend-psp.yaml

```diff
@@ -0,0 +1,35 @@
+{{- if and .Values.podSecurityPolicy.enabled .Values.defaultBackend.enabled -}}
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: {{ template "nginx-ingress.fullname" . }}-backend
+  labels:
+    app: {{ template "nginx-ingress.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+spec:
```

Comment:
For those who use a different default backend I am wondering if we should make this more customizable. Or at least provide a way to disable this. Thoughts?

Reply:
Right now the functionality is pretty flexible 

- if defaultBackend is disabled, none of these resources (Role, Rolebinding, PSP) will be created, so they can bring their own backend with any necessary PSPs, etc., they would just set `controller.defaultBackendService: <my custom backend>` and `defaultBackend.enabled: false`

- if RBAC is disabled, then the Role and RoleBindings will not be created, but Service Accounts will be created by default (which can be disabled/overridden for those that want to create their own).  This allows for users to BYO roles/policies with the defaultBackend.

- If PodSecurityPolicy is disabled, the result is the same as above - they're not created as they're only useful if you want our out of the box PSP.

Reply:
The only area with this PR that is mandatory changed is that the defaultBackend deployment pods now always needs to run with a Service Account.  TBH that's what it used to be before it was removed a few weeks ago because the controller's SA has elevated privileges. 

 I don't think it's a bad thing to bring it back, as it will be most common in prod clusters to need one for most deployments, and it can be overridden/renamed/bring-your-own.